src/Compilers/Core/MSBuildTask/Shared/ManagedCompiler.cs

```diff
@@ -922,6 +928,20 @@ bool resultFromHostObjectSetOperation
             }
         }
 
+        internal string InitializeHostObjectSupportForNewSwitches(ITaskHost hostObject, ref string param)
+        {
+            var compilerOptionsHostObject = hostObject as ICompilerOptionsHostObject;
+
+            if (compilerOptionsHostObject != null)
+            {
+                var commandLineBuilder = new CommandLineBuilderExtension();
+                AddResponseFileCommandsForSwitchesSinceInitialReleaseThatAreNeededByTheIDE(commandLineBuilder);
+                CheckHostObjectSupport(param = "CompilerOptions", compilerOptionsHostObject.SetCompilerOptions(commandLineBuilder.ToString()));
```

Comment:
This is the "pattern" for all the other CheckHostObjectSupport* calls in the codebase, for better or worse. I'm inclined to leave it as is and just rewrite the whole thing later.